### ISSUE TITLE
find() needs a select variable?

### ISSUE THREAD

**Issue by blyleven** [picqer/exact-php-client#34]
When I want to update a purchaseEntry with a newly uploaded document, I get this following error:

Error 400: Please add a $select or a $top=1 statement to the query string.

I'm using this lines to fetch the right purchaseEntry.

```
$purchaseInvoice	= new \Picqer\Financials\Exact\PurchaseEntry($connection);
$purchaseInvoice  	= $purchaseInvoice->find($purchaseInvoiceID);
$purchaseInvoice->Document = $documentID;
$purchaseInvoice->update();
```

When I change the find() function in Findable.php like this:
```
public function find($id, $select = '')
    {
        
        $result = $this->connection()->get($this->url, [
            '$filter' => $this->primaryKey . " eq guid'$id'",
            '$select' => $select
        ]);

        return new self($this->connection(), $result);
    }
```
Using this, it seems to work. 
```
$purchaseInvoice	= new \Picqer\Financials\Exact\PurchaseEntry($connection);
$purchaseInvoice  	= $purchaseInvoice->find($purchaseInvoiceID, 'EntryID, Document');
$purchaseInvoice->Document = $documentID;
$purchaseInvoice->update();
```

**Comment by stephangroen** [comment]
Strange, according to the docs just using an EntryID should work. I don't have any experience with PurchaseEntries myself.

You could use the filter method though, instead of an altered find() :

`$purchaseInvoice->filter("EntryID eq '$purchaseInvoiceID'", '', 'EntryID, Document');`

**Comment by blyleven** [comment]
Hmm strange, when I use the filter method, I get an 400 error:
Error 400: Operator 'eq' incompatible with operand types 'System.Guid' and 'System.String' at position 8.

When I do:
$purchaseInvoice->filter($purchaseInvoiceID, '', 'EntryID, Document');

This error comes up:
Error 400: No property 'c319fc7b' exists in type 'Exact.Web.Api.Models.PurchaseEntry' at position 0.
Don't know where c319fc7b comes from and it changes every request.

**Comment by stephangroen** [comment]
Fixed in #36 and v2.4.0
